src/synapse/workload.py

```diff
@@ -149,6 +162,19 @@ def check_mjolnir_ready() -> ops.pebble.CheckDict:
     return check.to_dict()
 
 
+def check_irc_bridge_ready() -> ops.pebble.CheckDict:
+    """Return the Synapse IRC bridge service check.
+
+    Returns:
+        Dict: check object converted to its dict representation.
+    """
+    check = Check(CHECK_IRC_BRIDGE_READY_NAME)
+    check.override = "replace"
+    check.level = "ready"
+    check.http = {"url": f"http://localhost:{IRC_BRIDGE_HEALTH_PORT}"}
+    return check.to_dict()
```

Comment:
This module probably shouldn't be `ops` aware - perhaps better to put this function into the layers that are `ops` / `juju` aware?

Reply:
Do you mind having a look at the whole file and perhaps make a proposal on where it is better to place ops aware functions? Reason is the module has lots of ops logic and if/when we refactor, I prefer this is done in one pass if possible. Thanks!

Reply:
You are right there is a lot of `ops` usage in here - that should all be moved out and these kind of modules shouldn't depend on `ops` and instead provide more generic functions and classes to support the charm

Reply:
I did a first pass and changed a lot of synapse config related to not depend on ops, please have a look. I propose to continue in another refactor PR, this will take time.